src/mrinufft/operators/autodiff.py

```diff
@@ -145,6 +136,19 @@ def adj_op(self, kspace):
         r"""Compute the adjoint k-space -> image."""
         return _NUFFT_ADJOP.apply(kspace, self.samples, self.nufft_op)
 
+    @property
+    def samples(self):
+        """Get the samples."""
+        try:
+            return self._samples_torch
+        except AttributeError:
+            return self.nufft_op.samples
+
+    @samples.setter
+    def samples(self, value):
+        self._samples_torch = value
+        self.nufft_op.samples = value.detach().cpu().numpy()
+
```

Comment:
I love this setter! It really avoids redundantly moving the device of the samples in the test!